lib/matplotlib/rcsetup.py

```diff
@@ -501,6 +501,8 @@ def __call__(self, s):
 
 # a map from key -> value, converter
 defaultParams = {
+    'style': ['', validate_stringlist],
```

Comment:
Here's a gist with a function that flattens a dictionary containing a special key pointing to "parent" dictionaries: https://gist.github.com/tonysyu/b02be7df210c4369afa6. That's a generic implementation with nothing specific to matplotlib styles. To use in this case you could write something like the following.

```python
PARENT_STYLES = 'styles'
flattened_style = flatten_style_dict({PARENT_STYLES: styles})

def flatten_style_dict(styles_dict):
    return flatten_inheritance_dict(styles_dict, PARENT_STYLES,
                                    expand_parent=_expand_parent)


def _expand_parent(parent_style):
    if cbook.is_string_like(parent_style):
        return get_style_parameters(parent_style)
    return parent_style
```

To actually integrate this with the `style` module, you'd want to remove the loop from `use` and just  call update on the `flattened_style` dictionary. Note that the current tests won't pass with this function because the order of styles in the `PARENT_STYLES` is flipped. If you flip the order of parent styles in your `mplstyle` files, the test will pass.